litestar/utils/dataclass.py

```diff
@@ -1,21 +1,54 @@
 from __future__ import annotations
 
-from dataclasses import Field, fields
+from dataclasses import Field, fields, is_dataclass
+from inspect import isclass
 from typing import TYPE_CHECKING
 
 from litestar.types import Empty
-from litestar.types.protocols import DataclassProtocol
 
 if TYPE_CHECKING:
     from typing import AbstractSet, Any, Iterable
 
+    from typing_extensions import TypeGuard
+
+    from litestar.types.protocols import DataclassProtocol
+
 __all__ = (
     "extract_dataclass_fields",
     "extract_dataclass_items",
+    "is_dataclass_class",
+    "is_dataclass_instance",
     "simple_asdict",
 )
 
 
+def is_dataclass_instance(obj: Any) -> TypeGuard[DataclassProtocol]:
```

Comment:
So, why not keep these inside the `predicates.py` namespace? It's where we have all type guards and similar predicates. I think consistency here is important.

Reply:
We have `utils.dataclass` so that is the place I expect that dataclass utils exist.

The same argument about consistency can be made about having dataclass utilities across two modules.

Reply:
There is precedent for predicates being grouped by functionality, e.g., `is_async_callable()` in the `utils.sync` module, `annotation_is_iterable_of_type()`  in `utils.typing`,

Reply:
Well, i don't see the need for this change. I expect predicates to be in predicates since that's what we have there. 

The other utils should be moved there too if anything.

Reply:
> The other utils should be moved there too if anything.

Created #1730